elytron/src/main/java/org/wildfly/extension/elytron/ExpressionResolverResourceDefinition.java

```diff
@@ -182,20 +185,49 @@ static ResourceDefinition getExpressionResolverDefinition(PathAddress parentAddr
 
         ElytronExpressionResolver expressionResolver = new ElytronExpressionResolver(
                 (e, c) -> configureExpressionResolver(resourceAddress, e, c));
-        RuntimeCapability<ExpressionResolver> expressionResolverRuntimeCapability =  RuntimeCapability
-                .Builder.<ExpressionResolver>of(EXPRESSION_RESOLVER_CAPABILITY, false, expressionResolver)
+        RuntimeCapability<ExpressionResolverExtension> expressionResolverRuntimeCapability =  RuntimeCapability
+                .Builder.<ExpressionResolverExtension>of(EXPRESSION_RESOLVER_CAPABILITY, false, expressionResolver)
                 .build();
 
         AbstractAddStepHandler add = new ExpressionResolverAddHandler(expressionResolverRuntimeCapability);
-        OperationStepHandler remove = new TrivialCapabilityServiceRemoveHandler(add, expressionResolverRuntimeCapability);
+        OperationStepHandler remove = new ExpressionResolverRemoveHandler(add, expressionResolverRuntimeCapability);
 
         return new ExpressionResolverResourceDefinition(add, remove, expressionResolverRuntimeCapability);
     }
 
     private static class ExpressionResolverAddHandler extends BaseAddHandler {
 
-        ExpressionResolverAddHandler(RuntimeCapability<ExpressionResolver> expressionResolverRuntimeCapability) {
+        private final ElytronExpressionResolver expressionResolver;
+
+        ExpressionResolverAddHandler(RuntimeCapability<ExpressionResolverExtension> expressionResolverRuntimeCapability) {
             super(expressionResolverRuntimeCapability, ATTRIBUTES);
+            expressionResolver = (ElytronExpressionResolver) expressionResolverRuntimeCapability.getRuntimeAPI();
+        }
+
+        @Override
+        protected void recordCapabilitiesAndRequirements(OperationContext context, ModelNode operation, Resource resource)
+                throws OperationFailedException {
+            super.recordCapabilitiesAndRequirements(context, operation, resource);
+        }
```

Comment:
Note that this looks like an add in this class but that's an artifact of the diffing algorithm; the block was already there but the diff puts the existing code block in the new ExpressionResolverRemoveHandler class.

That said I don't see the point of either of them. :)